Algorithm.CSharp/StochasticIndicatorWarmsUpProperlyRegressionAlgorithm.cs

```diff
@@ -0,0 +1,172 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Data;
+using QuantConnect.Data.Consolidators;
+using QuantConnect.Indicators;
+using QuantConnect.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// Regression algorithm that asserts Stochastic indicator, registered with a different resolution consolidator,
+    /// is warmed up properly by calling QCAlgorithm.WarmUpIndicator
+    /// </summary>
+    public class StochasticIndicatorWarmsUpProperlyRegressionAlgorithm: QCAlgorithm, IRegressionAlgorithmDefinition
+    {
+        private bool _dataPointsReceived;
+        private Symbol _spy;
+        private RelativeStrengthIndex _rsi;
+        private RelativeStrengthIndex _rsiHistory;
+        private Stochastic _sto;
+        private Stochastic _stoHistory;
+
+        public override void Initialize()
+        {
+            SetStartDate(2020, 1, 1);
+            SetEndDate(2020, 2, 1);
+
+            _spy = AddEquity("SPY", Resolution.Hour).Symbol;
+
+            var dailyConsolidator = new TradeBarConsolidator(TimeSpan.FromDays(1));
+            _rsi = new RelativeStrengthIndex(14, MovingAverageType.Wilders);
+            _sto = new Stochastic("FIRST", 14, 3, 3);
+            RegisterIndicator(_spy, _rsi, dailyConsolidator);
+            RegisterIndicator(_spy, _sto, dailyConsolidator);
+
+            WarmUpIndicator(_spy, _rsi, TimeSpan.FromDays(1));
+            WarmUpIndicator(_spy, _sto, TimeSpan.FromDays(1));
+
+            _rsiHistory = new RelativeStrengthIndex(14, MovingAverageType.Wilders);
+            _stoHistory = new Stochastic("SECOND", 14, 3, 3);
+            RegisterIndicator(_spy, _rsiHistory, dailyConsolidator);
+            RegisterIndicator(_spy, _stoHistory, dailyConsolidator);
+
+            var history = History(_spy, 15, Resolution.Daily);
```

Comment:
nit - for clarity, might use `Max(_rsi.WarmupPeriod, _sto.WarmupPeriod)` instead of `15`. And then tp update the indicators might use something like `history.TakeLast(indicator.WarmupPeriod)`